tempesta_fw/http.c

```diff
@@ -4102,7 +4229,13 @@ tfw_http_req_block(TfwHttpReq *req, int status, const char *msg)
 	tfw_http_cli_error_resp_and_log(req, status, msg, true, false);
 }
 
-
+/*
+ * TODO: http1 headers usually use mixed case, while h2 uses only lower case
+ * header names. It seems that usual browsers are OK with mixed case in h2
+ * responses, but this behaviour is not defined by RFC and implementation
+ * dependent. Curl is not happy with the mixed case and return parser errors.
+ * The headers should be converted to lower case before transmission.
+ */
```

Comment:
Oh, yes. I was confused by `Transfer-Encoding` filtered by curl. It wasn't because of the case, but because the header itself.

From the [RFC 7540 8.1.2](https://httpwg.org/specs/rfc7540.html#rfc.section.8.1.2)
 > Just as in HTTP/1.x, header field names are strings of ASCII characters that are compared in a case-insensitive fashion. However, header field names MUST be converted to lowercase prior to their encoding in HTTP/2. A request or response containing uppercase header field names MUST be treated as malformed (Section 8.1.2.6).

According the RFC an end point must close a connection (send error) on receiving our responses, but major browsers and curl _ignore_ this requirement. Well, it's a problem for us, actually. Converting header names to lower case definitely has performance impact. But even if browsers are OK, we can't just brake the RFC requirement and leave headers names in mixed case.